core/shared/src/main/scala/zio/logging/LogPattern.scala

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright 2019-2023 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.logging
+
+import zio.parser.{ Syntax, _ }
+import zio.{ Chunk, Config }
+
+import java.time.format.DateTimeFormatter
+import scala.util.Try
+
+/**
+ * A `LogPattern` is string representation of `LogFormat`
+ */
+sealed trait LogPattern {
+
+  /**
+   * Converts this log pattern into a log format.
+   */
+  def toLogFormat: LogFormat
+
+  def isDefinedFilter: Option[LogFilter[Any]] = None
+}
+
+object LogPattern {
+
+  sealed trait Arg extends LogPattern {
+    def name: String
+  }
+
+  sealed trait Arg1[A1] extends Arg {
```

Comment:
The type parameters here become erased because they are not tracked in the supertype.

Which means this is practically equivalent to `def arg1: Any`, etc. 

I wonder if, instead of doing it this way, we can either delete `Arg1` and `Arg2`, or embed type-safe capabilities into these treats, which lets clients of the traits use the arguments in some safe way?